temporian/implementation/numpy_cc/operators/tick_calendar_utils.cc

```diff
@@ -0,0 +1,54 @@
+#include "temporian/implementation/numpy_cc/operators/tick_calendar_utils.h"
+
+#include <cstdint>
+#include <optional>
+
+// Number of days in each month (non-leap year)
+constexpr int daysPerMonth[12] = {31, 28, 31, 30, 31, 30,
+                                  31, 31, 30, 31, 30, 31};
+
+bool IsLeapYear(int year) {
+  return ((year % 4) == 0) && ((year % 100) != 0 || (year % 400) == 0);
+}
+
+std::optional<MyTime> UTCMkTime(const int year, const int month, const int day,
+                                const int hour, const int minute,
+                                const int second) {
+  if (month < 1 || month > 12 || day > daysPerMonth[month - 1]) {
```

Comment:
IIUC, UTCMkTime(2000, 2, 29, 0, 0, 1) will return {}, which looks like a bug.

Reply:
*Solved + added more tests to check more similar cases.